### PR TITLE
fix(cloudflare): _routes.json deduplication

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
## Changes

- this should fix the deduplication error we had for more complex route definitions
- this is a fix for v9 of the adapter, in v10 we will refactor the complete route generation logic _(so this has a limited time to live)_
- also reverting change in test file https://github.com/withastro/adapters/pull/137/files#diff-d3130bfe10263d21a437f7e826fc4594ffc77e5cc37be1b09b10f5553b08c3ed
- closes https://github.com/withastro/adapters/issues/149

## Testing

- I personally don't think it is worth to add a complete fixture for such edge-case, however we can if anyone has strong opinions
- Let's investigate if we can add unit tests instead which test only util functions

### CODE REVIEW

**Comment by alexanderniebuhr** [comment]
Current dependencies on/for this PR:
* **#171** <a href="https://app.graphite.dev/github/pr/withastro/adapters/171?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/withastro/adapters/171?utm_source=stack-comment).

File: packages/cloudflare/src/utils/deduplicatePatterns.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function deduplicatePatterns(patterns: string[]) {
 	const uniquePatterns = [...new Set(patterns)];
 	for (const pattern of uniquePatterns) {
 		if (pattern.endsWith('*')) {
-			openPatterns.push(new RegExp(`^${pattern.replace(/(\*\/)*\*$/g, '[^*\n]*$')}`));
+			openPatterns.push(new RegExp(`^${pattern.replace(/(\*\/)*\*$/g, '(?=.{2,}).+[^*\n]$')}`));
```
Comment (ematipico) [inline packages/cloudflare/src/utils/deduplicatePatterns.ts:16]: What's the actual fix, though? The PR doesn't explain it

File: .changeset/sharp-cameras-rescue.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/cloudflare': patch
+---
+
+Fixes automatic deduplication of `_routes.json`, which had more complex project structures
```
Comment (sarah11918) [inline .changeset/sharp-cameras-rescue.md:5]:
```suggestion
Fixes an error with automatic deduplication of `_routes.json` for more complex project structures
```

**Review by sarah11918 — COMMENTED** [review]
Maybe something like this, if it's accurate?

**Review by ematipico — APPROVED** [review]

**Comment by MehrdadKhnzd** [comment]
Hi! It still doesn't work, I still get the error I used to get:
`Error: Failed to publish your Function. Got error: Error 8000057: Overlapping rules in _routes.json are not allowed.`

**Comment by MehrdadKhnzd** [comment]
@alexanderniebuhr The problem is that I couldn't. I tried to reproduce it two times, but I wasn't successful when I partially recreated some of the conflicting structures of my project. I even thought it might be due to some other added plugins, but that wasn't the case. I might give it another try with a complete replication of my file structure to see how it goes, but it will be really time-consuming. Do you know when v10 is going to be ready?
